app/views/ChargeSummary.scala.html

```diff
@@ -282,32 +237,32 @@ <h1 class="govuk-heading-xl">
         @p(){
             @if(viewModel.isAgent) {
                 @messages("chargeSummary.whatYouOwe.textOne-agent")
-                <a class="govuk-link" href="@whatYouOweUrl">@messages("chargeSummary.whatYouOwe.linkText-agent")</a>
+                <a class="govuk-link" href="@viewModel.whatYouOweUrl">@messages("chargeSummary.whatYouOwe.linkText-agent")</a>
                 @messages("chargeSummary.whatYouOwe.textTwo-agent")
             } else {
                 @messages("chargeSummary.whatYouOwe.textOne")
-                <a class="govuk-link" href="@whatYouOweUrl">@messages("chargeSummary.whatYouOwe.linkText")</a>
+                <a class="govuk-link" href="@viewModel.whatYouOweUrl">@messages("chargeSummary.whatYouOwe.linkText")</a>
                 @messages("chargeSummary.whatYouOwe.textTwo")
             }
         }
     }
 
-    @if(!viewModel.isAgent && documentDetail.remainingToPay > 0 && !(viewModel.codingOutEnabled && documentDetail.isPayeSelfAssessment)){
-        <div id="payment-link-@{taxYearTo}">
-            <a class="govuk-button" role="button" data-module="govuk-button" href="@controllers.routes.PaymentController.paymentHandoff(documentDetail.remainingToPay.toPence, viewModel.origin)">
+    @if(!viewModel.isAgent && viewModel.documentDetail.remainingToPay > 0 && !(viewModel.codingOutEnabled && viewModel.documentDetail.isPayeSelfAssessment)){
```

Comment:
if we can move this logic into viewModel as well ? it quite hard to understand what is going on here